### PR TITLE
Feature/room interface spells

### PR DESCRIPTION
**Pull request by gullskatten**

### CODE REVIEW

**Comment by gullskatten** [comment]
Spells are now retrieved from the database in the dialog!

File: android/src/main/kotlin/no/ntnu/beardblaster/dbclasses/SpellBook.kt [hunk 1]
```diff
@@ -6,17 +6,17 @@ import androidx.room.Junction
 import androidx.room.Relation
 
 @Entity(primaryKeys = ["wizardID", "spellID"])
-data class KnowsSpell(
+data class SpellBook(
```
Comment (cherrybeans) [inline android/src/main/kotlin/no/ntnu/beardblaster/dbclasses/SpellBook.kt:9]: Good change!

File: core/src/main/kotlin/no/ntnu/beardblaster/models/SpellCasting.kt [hunk 2]
```diff
@@ -53,8 +55,7 @@ class SpellCasting : Observer, Observable() {
 
     fun getSelectedSpell(): Spell? {
         if (null in selectedElements) return null
-        // Here you would query the database I guess to get data to initialize the spell model
-        return Spell(1, "Fireball", 12, "Blasts the enemy to pieces.")
+        return SpellRepository().getSpellById(selectedElements.map { elem -> elem?.elementID!! }.reduce { sum, element -> sum * element })
```
Comment (cherrybeans) [inline core/src/main/kotlin/no/ntnu/beardblaster/models/SpellCasting.kt:58]: Finally 😄

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/MenuScreen.kt [hunk 3]
```diff
@@ -76,6 +72,19 @@ class MenuScreen(
             add(exitBtn).colspan(2).center()
         }
         stage.addActor(table)
+
+        if (UserData.instance.user == null && !UserData.instance.isLoading) {
+            KtxAsync.launch {
+                UserData.instance.loadUserData()
+            }
+            UserData.instance.addObserver(this)
```
Comment (cherrybeans) [inline core/src/main/kotlin/no/ntnu/beardblaster/screen/MenuScreen.kt:80]: Is there a need for two UserData observers?

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/MenuScreen.kt [hunk 4]
```diff
@@ -76,6 +72,19 @@ class MenuScreen(
...
+        }
+        KtxAsync.launch {
+            UserData.instance.loadUserData()
```
Comment (cherrybeans) [inline core/src/main/kotlin/no/ntnu/beardblaster/screen/MenuScreen.kt]: And two loadUserDatas?

**Review by cherrybeans — APPROVED** [review]
After considering my comment its all good

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/MenuScreen.kt (hunk 3 above)
Comment (gullskatten) [reply core/src/main/kotlin/no/ntnu/beardblaster/screen/MenuScreen.kt:80]: øøøøø... no!

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/MenuScreen.kt (hunk 4 above)
Comment (gullskatten) [reply core/src/main/kotlin/no/ntnu/beardblaster/screen/MenuScreen.kt]: nope I don't think so!

**Pull request merged**